pkg/services/dashboards/service/integration_test/dashboard_k8s_test.go

```diff
@@ -0,0 +1,1680 @@
+				allowsEdits:   false,
+				shouldSucceed: false,
+			},
+			{
+				name:          "allow updating provisioned dashboard when allowsEdits is true",
+				allowsEdits:   true,
+				shouldSucceed: true,
+			},
+		}
+
+		for _, tc := range testCases {
+			t.Run(tc.name, func(t *testing.T) {
+				// Create a dashboard with admin
+				dash, err := createDashboard(t, adminClient, "Dashboard for Provisioning Test", nil, nil)
+				require.NoError(t, err, "Failed to create dashboard for provisioning test")
+				dashUID := dash.GetName()
+
+				// Fetch the created dashboard
+				fetchedDash, err := adminClient.Resource.Get(context.Background(), dashUID, v1.GetOptions{})
+				require.NoError(t, err)
+				require.NotNil(t, fetchedDash)
+
+				// Mark the dashboard as provisioned with allowsEdits parameter
+				provisionedDash := markDashboardObjectAsProvisioned(t, fetchedDash, "test-provider", "test-external-id", "test-checksum", tc.allowsEdits)
+
+				// Update the dashboard to apply the provisioning annotations
+				updatedDash, err := adminClient.Resource.Update(context.Background(), provisionedDash, v1.UpdateOptions{})
+				require.NoError(t, err)
+				require.NotNil(t, updatedDash)
+
+				// Re-fetch the dashboard after it's marked as provisioned
+				provisionedFetchedDash, err := editorClient.Resource.Get(context.Background(), dashUID, v1.GetOptions{})
+				require.NoError(t, err)
+				require.NotNil(t, provisionedFetchedDash)
+
+				// Try to update the dashboard using editor (not admin)
+				_, err = updateDashboard(t, editorClient, provisionedFetchedDash, "Updated Provisioned Dashboard")
+
+				if tc.shouldSucceed {
+					require.NoError(t, err, "Editor should be able to update provisioned dashboard when allowsEdits is true")
+
+					// Verify the update succeeded by fetching the dashboard again
+					updatedDash, err := editorClient.Resource.Get(context.Background(), dashUID, v1.GetOptions{})
+					require.NoError(t, err)
+					updatedSpec := updatedDash.Object["spec"].(map[string]interface{})
+					require.Equal(t, "Updated Provisioned Dashboard", updatedSpec["title"], "Dashboard title should be updated")
+				} else {
+					require.Error(t, err, "Editor should not be able to update provisioned dashboard when allowsEdits is false")
+					require.Contains(t, err.Error(), "provisioned")
+				}
+
+				// Clean up
+				err = adminClient.Resource.Delete(context.Background(), dashUID, v1.DeleteOptions{})
+				require.NoError(t, err)
+			})
+		}
+	})
+
+	t.Run("Dashboard refresh interval validations", func(t *testing.T) {
+		t.Run("reject dashboard with refresh interval below minimum", func(t *testing.T) {
+			dashObj := createDashboardObject("Dashboard with Short Refresh", "", 0)
+
+			// Add refresh configuration that should be too frequent
+			if spec, ok := dashObj.Object["spec"].(map[string]interface{}); ok {
+				spec["refresh"] = "1s" // Too frequent, below typical minimum
```

Comment:
that can be set in the config.ini with 
[dashboards]
min_refresh_interval = <time>

Reply:
TIL, thanks! I'm now setting a specific value before the test.